robot_framework/eflyt.py

```diff
@@ -0,0 +1,338 @@
+
+    if len(queue_elements) == 0:
+        return True
+
+    # If the case has been tried more than once before skip it
+    if len(queue_elements) > 1:
+        orchestrator_connection.log_info("Skipping: Case has failed in the past.")
+        return False
+
+    # If it has been marked as done, skip it
+    if queue_elements[0].status == QueueStatus.DONE:
+        orchestrator_connection.log_info("Skipping: Case already marked as done.")
+        return False
+
+    return True
+
+
+def count_beboere(browser: webdriver.Chrome) -> int:
+    """Count the number of beboere living on the address.
+
+    Args:
+        browser: The webdriver browser object.
+
+    Returns:
+        The number of beboere on the address.
+    """
+    change_tab(browser, 1)
+    beboer_table = browser.find_element(By.ID, "ctl00_ContentPlaceHolder2_ptFanePerson_becPersonTab_GridViewBeboere")
+    rows = beboer_table.find_elements(By.TAG_NAME, "tr")
+
+    # Remove header
+    rows.pop(0)
+
+    return len(rows)
+
+
+def get_room_count(browser: webdriver.Chrome) -> int:
+    """Get the number of rooms on the address.
+
+    Args:
+        browser: The webdriver browser object.
+
+    Returns:
+        The number of rooms on the address.
+    """
+    area_room_text = browser.find_element(By.ID, "ctl00_ContentPlaceHolder2_ptFanePerson_stcPersonTab6_lblAreaText").text
+    room_text = area_room_text.split("/")[1]
+    return int(room_text)
+
+
+def get_applicants(browser: webdriver.Chrome) -> list[str]:
+    """Get a list of applicants' cpr numbers from the applicant table.
+
+    Args:
+        browser: The webdriver browser object.
+
+    Returns:
+        A list of cpr numbers.
+    """
+    table = browser.find_element(By.ID, "ctl00_ContentPlaceHolder2_GridViewMovingPersons")
+    rows = table.find_elements(By.TAG_NAME, "tr")
+
+    # Remove header row
+    rows.pop(0)
+
+    cpr_list = []
+
+    for row in rows:
+        cpr = row.find_element(By.XPATH, "td[2]/a[2]").text
+        cpr_list.append(cpr)
+
+    return cpr_list
+
+
+def approve_case(browser: webdriver.Chrome):
+    """Approve the case and add a note about it.
+
+    Args:
+        browser: _description_
+    """
+    change_tab(browser, 0)
+
+    # Set note
+    create_note(browser, f"{date.today()} Besked fra Robot: Automatisk godkendt.")
+
+    # Click 'Godkend'
+    browser.find_element(By.ID, "ctl00_ContentPlaceHolder2_ptFanePerson_stcPersonTab1_btnGodkend").click()
+
+    # Click 'OK' in popup
+    browser.find_element(By.ID, "ctl00_ContentPlaceHolder2_ptFanePerson_stcPersonTab1_btnApproveYes").click()
+
+    # Click 'Godkend' personer
+    browser.find_element(By.ID, "ctl00_ContentPlaceHolder2_ptFanePerson_stcPersonTab1_btnGodkendAlle").click()
+
+
+def get_age(cpr: str) -> int:
+    """Get the age of a person based on their cpr number
+    assuming they are between 0-99 years old.
```

Comment:
The 7th digits in the CPR tells in which century a person is born, cf. https://cpr.dk/cpr-systemet/opbygning-af-cpr-nummeret which in turn refers to https://cpr.dk/media/12066/personnummeret-i-cpr.pdf.

Reply:
🤯